src/assets/scripts/game/GameController.js

```diff
@@ -68,107 +71,61 @@ export default class GameController {
      */
     init_pre() {
         prop.game = game;
-        // TODO: with the assignment of `prop.game = game`, the rest of these assignments are redundant and can
-        // be removed after proper testing
-        prop.game.paused = true;
-        prop.game.focused = true;
-        prop.game.speedup = 1;
-        prop.game.frequency = 1;
-        prop.game.time = 0;
-        prop.game.delta = 0;
-        prop.game.timeouts = [];
 
+        // Set blurring function
         $(window).blur(() => {
             prop.game.focused = false;
         });
 
+        // Set un-blurring function
         $(window).focus(() => {
             prop.game.focused = true;
         });
 
-        prop.game.last_score = 0;
-        prop.game.score = {
-            arrival: 0,
-            departure: 0,
-
-            windy_landing: 0,
-            windy_takeoff: 0,
-
-            failed_arrival: 0,
-            failed_departure: 0,
-
-            warning: 0,
-            hit: 0,
-
-            abort: {
-                landing: 0,
-                taxi: 0
-            },
-
-            violation: 0,
-            restrictions: 0
-        };
-
-        prop.game.option = new GameOptions();
+        // Initialize event count
+        _forEach(GAME_EVENTS, (gameEvent, key) => {
+            this.game.events[key] = 0;
+        });
     }
 
     /**
      * @for GameController
-     * @method game_get_score
+     * @method game_get_weighted_score
      */
-    game_get_score() {
-        let score = 0;
-
-        score += prop.game.score.arrival * 10;
-        score += prop.game.score.departure * 10;
-
-        score -= prop.game.score.windy_landing * 0.5;
-        score -= prop.game.score.windy_takeoff * 0.5;
-
-        score -= prop.game.score.failed_arrival * 20;
-        score -= prop.game.score.failed_departure * 2;
-
-        score -= prop.game.score.warning * 5;
-        score -= prop.game.score.hit * 50;
-
-        score -= prop.game.score.abort.landing * 5;
-        score -= prop.game.score.abort.taxi * 2;
-
-        score -= prop.game.score.violation;
-        score -= prop.game.score.restrictions * 10;
+    game_get_weighted_score() {
+        const hoursPlayed = this.game_time() / TIME.ONE_HOUR_IN_SECONDS;
+        const scorePerHour = this.game.score / hoursPlayed;
 
-        return score;
+        return scorePerHour;
     }
 
     /**
+     * Record a game event to this.game.events, and update this.game.score
      * @for GameController
-     * @method game_get_weighted_score
+     * @method record_game_event
+     * @param gameEvent {String} one of the events listed in GAME_EVENTS
      */
-    game_get_weighted_score() {
-        let score = this.game_get_score();
-
-        score = score / (this.game_time() / 60);
-        score *= 500;
+    record_game_event(gameEvent) {
```

Comment:
👍 👍 
For abstracting this to a method _and_ centralizing where game events are updated